frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -316,6 +316,10 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 		}
 
 		if v := st.Platform; v != "" {
+			// Mark whether the specified platform is redudant for the linter
+			// to pick up later.
+			ds.platformRedundant = v == "$TARGETPLATFORM"
+
```

Comment:
If we are going to defer returning the warning to dispatch() (to only return warnings for reachable stages) then it should be done for this `reportUnusedFromArgs` condition as well. In that case the key in the `dispatchState` can be something like `stageWarnings` or `warnings` and contain the slice of warnings that will all be reported if stage is found to be reachable.

Reply:
I've integrated this with the existing report unused args using the same shell match and reported the error there. If we decide that warnings should only be reported if reachable, we can change that for all of them at once. I'd like to avoid modifying how the other variables are reported within this PR.